### PR TITLE
[core] add serve chaos test

### PR DESCRIPTION
**Pull request by rynewang**
Adds CI chaos test in a more realistic workload.

- Workload: Ray Serve running mock LLM; client sending HTTP request to query.
- Chaos: 200ms network delay.

### CODE REVIEW

File: .buildkite/pipeline.build.yml [hunk 1]
```diff
@@ -382,6 +382,57 @@
...
+    # Helm install chaos-mesh
+    - echo "--- Installing chaos-mesh operator and CR."
+    - helm repo add chaos-mesh https://charts.chaos-mesh.org
+    - kubectl create ns chaos-mesh
+    - helm install chaos-mesh chaos-mesh/chaos-mesh -n=chaos-mesh --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock --version 2.6.1
+    - kubectl wait pod  --namespace chaos-mesh  -l app.kubernetes.io/instance=chaos-mesh --for=condition=Ready=True
+    - echo "--- Running the script without faults"
+    - ray job submit --address http://localhost:8265 --runtime-env python/ray/tests/chaos/streaming_llm.yaml --working-dir python/ray/tests/chaos -- python streaming_llm.py --num_queries_per_task=100 --num_tasks=2 --num_words_per_query=100
+    # Now add the delay, rerun the job
+    - kubectl apply -f python/ray/tests/chaos/chaos_network_delay.yaml
```
Comment (rkooo567) [inline .buildkite/pipeline.build.yml:431]: Can we make the common config to the script from other CI test?

File: python/ray/tests/chaos/streaming_llm.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        "--num_words_per_query",
+        type=int,
+        required=True,
+        help="Number of words per query",
+    )
+
+    args = parser.parse_args()
+
+    # Run the serve, run the client, then showdown serve.
+    llm = ReverseAndDupEachWord.bind(2)
```
Comment (rkooo567) [inline python/ray/tests/chaos/streaming_llm.py:92]: can you make dup_time configurable and make it a bit longer? (50?) So the output is more realisitc (I think in this case 50 * 26 bytes == 1~ishKB?)

File: .buildkite/pipeline.build.yml [hunk 3]
```diff
@@ -382,6 +382,57 @@
...
+    - echo "--- Running the script with fault of networking delay"
+    - ray job submit --address http://localhost:8265 --runtime-env python/ray/tests/chaos/streaming_llm.yaml --working-dir python/ray/tests/chaos -- python streaming_llm.py --num_queries_per_task=100 --num_tasks=2 --num_words_per_query=100
```
Comment (rkooo567) [inline .buildkite/pipeline.build.yml:433]: How long does it usually run?

**Review by rkooo567 — APPROVED** [review]
LGTM. Can you also ping someone else to review if the LLM mimicking code is realistic (maybe from the Aviary team)?

Also, do you plan to create a release test that runs for a day?

File: .buildkite/pipeline.build.yml (hunk 1 above)
Comment (rynewang) [reply .buildkite/pipeline.build.yml:431]: yes, but maybe in next PR. I will make it a matrix: `workload x chaos`

File: python/ray/tests/chaos/streaming_llm.py (hunk 2 above)
Comment (rynewang) [reply python/ray/tests/chaos/streaming_llm.py:92]: ok, working on it

File: .buildkite/pipeline.build.yml (hunk 3 above)
Comment (rynewang) [reply .buildkite/pipeline.build.yml:433]: See https://buildkite.com/ray-project/oss-ci-build-pr/builds/33173#018a1eea-a5f7-4518-8ac7-bb294c34f7bb.

No chaos: 1m25s
With delay: 3m38s but can run to completion

File: python/ray/tests/chaos/streaming_llm.py (hunk 2 above)
Comment (rynewang) [reply python/ray/tests/chaos/streaming_llm.py:92]: on second thought I think we can keep it as it. The workload size can be tweaked with flags `num_words_per_query`

Comment (rkooo567) [reply python/ray/tests/chaos/streaming_llm.py:92]: Let's use more realistic queries in release tests then

**Comment by rkooo567** [comment]
cc @zhe-thoughts this is test that's needed for multi cloud work, and it is test-only changes (low risk)

**Review by zhe-thoughts — APPROVED** [review]
+1 to @rkooo567

**Pull request merged**